detections/endpoint/executables_or_script_creation_in_temp_path.yml

```diff
@@ -0,0 +1,116 @@
+name: Executables Or Script Creation In Temp Path
+id: e0422b71-2c05-4f32-8754-01fb415f49c9
+version: 11
+date: '2025-02-11'
+author: Teoderick Contreras, Splunk
+status: production
+type: Anomaly
+description: The following analytic identifies the creation of executables or scripts
+  in suspicious file paths on Windows systems. It leverages the Endpoint.Filesystem
+  data model to detect files with specific extensions (e.g., .exe, .dll, .ps1) created
+  in uncommon directories (e.g., \windows\fonts\, \users\public\). This activity is
+  significant as adversaries often use these paths to evade detection and maintain
+  persistence. If confirmed malicious, this behavior could allow attackers to execute
+  unauthorized code, escalate privileges, or persist within the environment, posing
+  a significant security threat.
+data_source:
+- Sysmon EventID 11
+search: '| tstats `security_content_summariesonly` values(Filesystem.file_path) as file_path count min(_time) as firstTime max(_time) as lastTime from datamodel=Endpoint.Filesystem
+  where Filesystem.file_name IN ("*.exe", "*.dll", "*.sys", "*.com", "*.vbs", "*.vbe", "*.js", "*.ps1", "*.bat", "*.cmd", "*.pif") 
+  AND Filesystem.file_path IN ("*\\temp\\*")
```

Comment:
Its best to not do a broad `temp` as this is a common name and you could match on unintended folders.

Here are some examples of some temp folders

- `C:\Windows\WinSxS\Temp\`
- `C:\$WINDOWS.~BT\NewOS\Windows\WinSxS\Temp\`
- `C:\Program Files\Google\Chrome\Temp\`
- `C:\ProgramData\Cisco\Cisco Secure Client\Temp\`
- `C:\Windows\ServiceProfiles\NetworkService\AppData\Local\Temp\`

I suggest focusing on `\AppData\Local\Temp`, `:\Windows\Temp\` and `:\Temp` as those are the most common ones. And leave the rest of the coverage to the users by indicating this somewhere in the description.

Reply:
good point will update this one